ydb/core/tx/datashard/ut_common/datashard_ut_common.cpp

```diff
@@ -2459,6 +2459,73 @@ namespace {
         TDeque<TPendingRequest> QuotaRequests;
     };
 
+    std::unique_ptr<TEvDataShard::TEvReadResult> ReadAllTable(
+        Tests::TServer::TPtr server,
+        ui64 tabletId,
+        const TTableId& tableId,
+        const NKikimrTxDataShard::TEvGetInfoResponse::TUserTable& userTable,
+        ui64 readId)
+    {
+        auto request = std::make_unique<TEvDataShard::TEvRead>();
+        auto& record = request->Record;
+
+        record.SetReadId(readId);
+        record.MutableTableId()->SetOwnerId(tableId.PathId.OwnerId);
+        record.MutableTableId()->SetTableId(userTable.GetPathId());
+
+        const auto& description = userTable.GetDescription();
+        for (const auto& column: description.GetColumns()) {
+            record.AddColumns(column.GetId());
+        }
+
+        // read all
+        auto fromBuf = TSerializedCellVec::Serialize(TVector<TCell>());
+        auto toBuf = TSerializedCellVec::Serialize(TVector<TCell>());
+        request->Ranges.emplace_back(fromBuf, toBuf, true, true);
+
+        record.SetResultFormat(NKikimrDataEvents::FORMAT_CELLVEC);
+
+        auto& runtime = *server->GetRuntime();
+        auto sender = runtime.AllocateEdgeActor();
+
+        runtime.SendToPipe(
+            tabletId,
+            sender,
+            request.release()
+        );
+
+        TAutoPtr<IEventHandle> handle;
+        runtime.GrabEdgeEventRethrow<TEvDataShard::TEvReadResult>(handle, TDuration::Seconds(5));
```

Comment:
There should be something like that
https://github.com/ydb-platform/ydb/blob/main/ydb/core/tx/datashard/datashard_ut_read_iterator.cpp#L560

Reply:
Yes, I suggested to move this function to common code

Reply:
By the way, you can move functions like `WaitReadResult` from `datashard_ut_read_iterator.cpp` to `datashard_ut_common.cpp`

Reply:
I set a limited timeout to prevent slowing down the test run in case of problems affecting all tests. However, this timeout would not be enough when running tests with profilers with significant performance overhead, such as Valgrind. So, I'll remove timeout.